provider/storage_benchmark.py

```diff
@@ -0,0 +1,322 @@
+        self.env_files.append(dst)
+
+    def run(self, cmd, timeout=600):
+        """
+        Execute the benchmark command.
+
+        :param cmd: executed command
+        :type cmd: str
+        :param timeout: timeout for executing command
+        :type timeout: float
+        :return: output of running command
+        :rtype: str
+        """
+        return self.session.cmd(cmd, timeout=timeout)
+
+    def clean(self, timeout=1800, force=False):
+        """
+        Clean benchmark tool packages and processes after testing inside guest.
+
+        :param timeout: timeout for cleaning
+        :type timeout: float
+        :param force: if is True, kill the running processes
+                      by force, otherwise wait they are done
+        :type force: bool
+        """
+        if force:
+            self._kill_procs()
+        else:
+            self._wait_procs_done(timeout)
+        self._remove_env_files()
+
+
+class IozoneCfg(object):
+    def __init__(self, os_type, params, instance):
+        if os_type == 'linux':
+            version = params.get('iozone_version', 'iozone3_483')
+            self.download_url = ('http://www.iozone.org/src/current/%s.tar' % version)
+            self.download_path = os.path.join('/home', 'iozone.tar')
+            self.iozone_dir = os.path.join('/home/iozone_inst', version)
+            self.arch = 'linux-AMD64' if 'x86_64' in machine() else 'linux-powerpc64'
+            self.cmd = 'cd %s/src/current && make %s' % (self.iozone_dir, self.arch)
+            self.link_src = '%s/src/current/iozone' % self.iozone_dir
+            self.iozone_path = '/usr/local/bin/iozone'
+            self.setups = {'download_benchmark': (CURL_DOWNLOAD,
+                                                  self.download_url,
+                                                  self.download_path),
+                           'unpack_file': (TAR_UNPACK, self.download_path,
+                                           '/home/iozone_inst'),
+                           'session_cmd': (self.cmd, 300),
+                           'make_symlinks': (self.link_src, self.iozone_path)}
+        else:
+            label = params.get('win_utils_label', 'WIN_UTILS')
+            drive_letter = utils_misc.get_winutils_vol(instance.session, label)
+            self.cmd = 'set nodosfilewarning=1 && set CYGWIN=nodosfilewarning'
+            self.iozone_path = drive_letter + r':\Iozone\iozone.exe'
+            self.setups = {'session_cmd': (self.cmd, 300)}
+
+
+class Iozone(StorageBenchmark):
+    def __init__(self, params, session):
+        super(Iozone, self).__init__(params['os_type'], session, 'iozone')
+        self.cfg = IozoneCfg(params['os_type'], params, self)
+        for method, args in self.cfg.setups.items():
+            getattr(self, method)(*args)
+
+    def run(self, cmd_options='-a', timeout=1800):
+        """
+        Run iozone test inside linux guest.
```

Comment:
maybe typo, should support both windows and linux.

Reply:
thanks, have updated.